### PR TITLE
Handle wait_for_deployment_completion during Azure deployment.

### LINKED ISSUE #26014: azure_rm_deployment ignores 'wait_for_deployment_completion' option
##### ISSUE TYPE
Bug Report

##### COMPONENT NAME
azure_rm_deployment

##### ANSIBLE VERSION
```
2.3.1.0
```

##### CONFIGURATION
*nope*

##### OS / ENVIRONMENT

macOS Sierra

##### SUMMARY

The `azure_rm_resourcegroup` module ignores the `wait_for_deployment_completion: False` option and blocks until the deployment completes. 

##### STEPS TO REPRODUCE

1. Download the [example ARM templates](https://github.com/Azure/azure-quickstart-templates/tree/master/101-container-registry) to the local ansible directory.

2. Run the following playbook.

```yaml
- name: Create infra
  hosts: localhost
  gather_facts: False

  tasks:

    - name: Create resource group
      azure_rm_resourcegroup:
        name: foobar
        location: westeurope
      register: reg_rg

    - name: Deploy Docker registry
      azure_rm_deployment:
        deployment_name: registry
        resource_group_name: '{{ reg_rg.state.name }}'
        location: '{{ reg_rg.state.location }}'
        template: "{{ lookup('file', './azuredeploy.json') }}"
        parameters: "{{ lookup('file', './azuredeploy.parameters.json') }}"
        wait_for_deployment_completion: False
      register: reg_depl

    - debug: var=reg_depl
```

##### EXPECTED RESULTS
Ansible initiates the deployment then moves on to the `debug` task without waiting for the deployment completion.

##### ACTUAL RESULTS
Ansible blocks until the deployment has completed.

### PR DESCRIPTION
**Pull request by rajasaur**
###### SUMMARY
Pull the get_poller_result inside the if block so that if the caller has
wait_for_deployment_completion=False, it doesnt block and wait for it to
finish.

Also, since the result contains information about the deployment, provide
None values for it in the output.(Not sure if this needs to be documented)

Fixes #26014 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
cloud/azure

##### ANSIBLE VERSION
```
ansible 2.4.0 (bug_26014 ab7059b33a) last updated 2017/06/26 13:46:21 (GMT +550)
  config file = None
  configured module search path = ['/Users/raja/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /private/tmp/ansible/lib/ansible
  executable location = /private/tmp/ansible/bin/ansible
  python version = 3.6.0 (default, Dec 30 2016, 23:33:29) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.42.1)]
```

##### ADDITIONAL INFORMATION
Since the call is now non-blocking the output has been changed to provide None values for deployment id, instances and properties.

### CODE REVIEW

**Comment by rajasaur** [comment]
The error from Shippable seems unrelated ?
https://app.shippable.com/github/ansible/ansible/runs/26490/4/console

**Comment by lmazuel** [comment]
The code itself LGTM (FYI @tstringer)

**Comment by trstringer** [comment]
shipit

**Comment by trstringer** [comment]
@lmazuel thanks for looping me in! If it looks good to you, we can auto-merge this by having a quorum of `shipit` comments here for ansibot.

**Comment by trstringer** [comment]
bot_status

**Comment by devigned** [comment]
shipit

**Comment by trstringer** [comment]
bot_status

**Comment by lmazuel** [comment]
shipit

**Comment by trstringer** [comment]
This module is supported_by `curated` so we won't be able to auto-merge.

@nitzmahone can you check this out and merge if you see fit?

**Comment by 4c74356b41** [comment]
@nitzmahone any chance to merge this?

**Comment by 4c74356b41** [comment]
@tstringer hey there, you seem to be pretty active here. can you push (i mean talk to a person to merge this) this as it appears done?

**Comment by trstringer** [comment]
Unfortunately I don't have permissions to merge! @nitzmahone would be the one to make the final call on this one!

If you are so inclined to have a direct conversation with the team, please feel free to join the IRC channel `#ansible-devel` or more specifically `#ansible-azure` on freenode. I know they are extremely busy at the moment, and I have no doubt this is on @nitzmahone's backlog.

Thanks so much for keeping on top of this, @4c74356b41! 😄 👍

**Comment by trstringer** [comment]
@4c74356b41 I should have mentioned this as well. Weekly there are a couple of core meetings on IRC (`#ansible-meeting`) that you can attend to ask for specific status. This may be a good one to join up for and bring to their attention.

The meetings are a good time to have the core Ansible team's undivided attention. Just a suggestion!

**Comment by nitzmahone** [comment]
@tstringer Code looks fine on inspection, but I'd like to hold this for the CI stuff that's happening right now so we can have a good set of PRs to exercise that with. I'm adding the "needs_revision" label just to prevent anyone from merging it (no revisions are actually necessary).

**Comment by trstringer** [comment]
@nitzmahone that sounds perfect to me! Agreed.

**Comment by 4c74356b41** [comment]
@nitzmahone hey there, any eta on this? as I don't know your internal stuff, what you said makes no sense to me, could you clarify, please? Thanks!

ps. @tstringer or maybe you could? :)

**Comment by nitzmahone** [comment]
I think the plan is to basically get all the pent-up Azure stuff merged before 2.4 module freeze on August 29. @tstringer is working on getting automated testing for Azure modules plugged into our build infrastructure so we can verify that things are more-or-less working without manually standing up a bunch of stuff to test against. I think it's getting really close.

**Comment by 4c74356b41** [comment]
@nitzmahone hey there, I saw you pushed 2.0.0, any timeline for this thing?

**Review by nitzmahone — APPROVED** [review]

**Comment by abadger** [comment]
Cherry-picked for the 2.4.0rc3 release.

**Comment by 4c74356b41** [comment]
@abadger howdy, do these rc releases get distributed via package management systems like pip?

**Comment by abadger** [comment]
Currently they aren't being distributed in a way that pip picks up
automatically.  You can install via pip, though, if you point it at the URL
for the release.  They're all available from this directory:.
http://releases.ansible.com/ansible/

**Comment by 4c74356b41** [comment]
@abadger that's really cool. any eta on 2.4.0rc3?

**Comment by abadger** [comment]
Later today.  I had an unexpected dental visit.  I'll be cutting the
2.4.0rc3 release soon after I get back from that.

**Pull request merged**
